includes/embeds/class-amp-reddit-embed-handler.php

```diff
@@ -0,0 +1,85 @@
+<?php
+/**
+ * Class AMP_Reddit_Embed_Handler
+ *
+ * @package AMP
+ * @since 0.7
+ */
+
+/**
+ * Class AMP_Reddit_Embed_Handler
+ */
+class AMP_Reddit_Embed_Handler extends AMP_Base_Embed_Handler {
+	/**
+	 * Regex matched to produce output amp-reddit.
+	 *
+	 * @const string
+	 */
+	const URL_PATTERN = '#https?://(www\.)?reddit\.com/r/[^/]+/comments/.*#i';
+
+	/**
+	 * Default height.
+	 *
+	 * @var int
+	 */
+	protected $DEFAULT_HEIGHT = 141; // phpcs:ignore WordPress.NamingConventions.ValidVariableName.MemberNotSnakeCase
+
+	/**
+	 * Default width.
+	 *
+	 * @var int
+	 */
+	protected $DEFAULT_WIDTH = 596; // phpcs:ignore WordPress.NamingConventions.ValidVariableName.MemberNotSnakeCase
```

Comment:
Maybe the oEmbed handler can make a request off to a URL like https://www.reddit.com/comments/7t6msa.json?limit=1&app=embed to obtain the `width`/`height` specific for the given Reddit embed.

Reply:
I've gone ahead and made it follow 100%/100% for the responsive layout type. This solves longer embeds using media that were getting cut off by the fixed dimensions, but now suffers from the same issue as reported [here](https://github.com/ampproject/amphtml/issues/11869).

The dimensions given in https://www.reddit.com/comments/7t6msa.json?limit=1&app=embed are just for the media attachments and wouldn't encompass the entire embed content.